### ISSUE TITLE
Compatibility with HashWithIndifferentAccess is broken in 0.10.3

### ISSUE THREAD

**Issue by kirs** [dry-rb/dry-validation#271]
```ruby
require 'bundler/setup'
require 'active_support/hash_with_indifferent_access'
require 'dry-validation'
params = ActiveSupport::HashWithIndifferentAccess.new(name: "kir", unknown: "kir")

form = Dry::Validation.Form do
  required(:name).filled
end
puts form.call(params).output.inspect

params = { name: "kir", unknown: "kir" }
puts form.call(params).output.inspect
```

Running on `dry-validation 0.9.3` (valid):

Output:

```
{:name=>"kir"}
{:name=>"kir"}
```

Output running on `dry-validation (0.10.3)` (invalid):

```
{"name"=>"kir", "unknown"=>"kir"}
{:name=>"kir"}
```

As you see, on `0.10.3` it keeps the `unknown` key when it's supposed to be drop it because that key is not specified in the schema.

cc @solnic

**Comment by solnic** [comment]
We have a longish thread in another issue about that: https://github.com/dry-rb/dry-validation/issues/262

Quick workaround for now: `params.permit!`

We're gonna address this one way or the other in the next release.

**Comment by kirs** [comment]
I'm talking about `HashWithIndifferentAccess`, not `AC::Parameters`. It doesn't have a `permit!` method.

You probably had no goal to support `HashWithIndifferentAccess`, but it worked in previous versions and we've used that compatibly.

**Comment by solnic** [comment]
I released dry-types 0.9.1 which brings back previous behavior and accepts Hash-subclasses too

**Comment by kirs** [comment]
Thanks! ❤️ ❤️ ❤️
